### PR TITLE
Don't treat precondition failures as distributed tracing failures

### PR DESCRIPTION
**Pull request by tg-msft**
We're allow listing a few select error codes for precondition failures that are typically considered expected behavior and we'd prefer they not show up as errors in Azure Monitor.

This (obviously) breaks a bunch of the Access Conditions tests.  I'm not going to patch those up unless we decide we want to move forward with this approach.

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/PartitionedUploader.cs [hunk 1]
```diff
@@ -227,7 +227,7 @@ internal class PartitionedUploader
                     accessTier,
                     cancellationToken);
             }
-            catch (Exception ex)
+            catch (Exception ex) when (BlobErrors.IsDiagnosticFailure(ex))
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Blobs/src/PartitionedUploader.cs]: Wonder if we can put this logic onto the ClientDiagnostics

Comment (tg-msft) [reply sdk/storage/Azure.Storage.Blobs/src/PartitionedUploader.cs]: I think the logic is pretty Blobs specific.  If you want to have some other way of hooking it into `ClientDiagnostics` (maybe overriding a method in a derived `BlobsClientDiagnostics`), that could work too.

**Comment by pakrym** [comment]
There is a lot of discussion currently how do we want to represent these scenarios in opentelemetry so I wouldn't spend time on implementation until those settle down.

**Comment by tg-msft** [comment]
That sounds fine - I'll close this PR for now and let you solve it across services when there's more clarity.

**Pull request closed without merging**